hs_collection_resource/utils.py

```diff
@@ -148,6 +149,17 @@ def get_collectable_resources(user, coll_resource):
     collectable_resources = collectable_resources.only('short_id', 'title', 'resource_type', 'created')
     collectable_resources = collectable_resources.select_related('raccess')
 
+    # fetch the owners of the resources rather than query for each resource
+    collectable_resources_dict = {r.short_id: r for r in collectable_resources}
+    collectable_resource_ids = collectable_resources_dict.keys()
+    urp_qs = UserResourcePrivilege.objects \
+        .filter(resource__short_id__in=collectable_resource_ids, privilege=PrivilegeCodes.OWNER)\
+        .select_related("user", "resource") \
+        .distinct()
+    for urp in urp_qs:
```

Comment:
I was initially doing this for loop to set the owner for each of the resources. Then I realized we can further speedup, by passing the urp_qs (the list of UserResourcePrevilege) to the UI and access the resource and user from each of the UserResourcePrevilege object in the template.

Reply:
good call! I'll add that

Reply:
I reverted the `get_collactable_resources` and instead added the `UserResourcePrevilege` directly into the `get_collectable_resources_modal` view

Let me know what you think and thank you for all your help!